src/components/PostsPage/PostsPage.vue

```diff
@@ -0,0 +1,246 @@
+  sidebarStatus.value = null;
+};
+
+const closePostEditing = () => {
+  if (sidebarStatus.value === "Post Editing") {
+    return (sidebarStatus.value = "Post");
+  }
+
+  sidebarStatus.value = null;
+};
+
+const openCreateNewPost = () => {
+  chosenPost.value = null;
+  sidebarStatus.value = "Post Adding";
+  postComments.value = null;
+};
+
+const editPost = () => {
+  sidebarStatus.value = "Post Editing";
+};
+
+const submitPostChange = async ({ id, userId, title, body }: Post) => {
+  if (sidebarStatus.value === "Post Editing") {
+    try {
+      const updatedPost = await patchPost({ id, title, body });
+
+      postList.value = postList.value.map((el) => {
+        if (el.id === id) {
+          return { ...el, title, body };
+        }
+
+        return el;
+      });
+      changeChosenPost(updatedPost?.id!);
+    } catch {
+      console.error("Failed to update post");
+    } finally {
+      sidebarStatus.value = "Post";
+      return;
+    }
+  }
+
+  try {
+    const addedPost = await createNewPost({ userId, title, body });
+
+    postList.value = [...postList.value, addedPost];
+    changeChosenPost(addedPost?.id!);
+  } catch {
+    console.error("Failed to add new post");
+  }
+};
+
+const changeCommentAddingState = () => {
+  isCommentsEditing.value = !isCommentsEditing.value;
+};
+
+const addNewComment = async ({ postId, name, body, email }: Comment) => {
+  try {
+    const newComment = await createComment({ postId, name, body, email });
+
+    postComments.value = [...postComments.value!, newComment];
+    isCommentsEditing.value = false;
+  } catch {
+    console.error("failed to added comment");
+  }
+};
+
+const deletePostComment = async (commentId: number) => {
+  try {
+    await deleteComment(commentId);
+
+    postComments.value = postComments.value!.filter((el) => el.id !== commentId);
+  } catch {
+    console.error("Failing to delete comment");
+  }
+};
+</script>
+
+<template>
+  <Header :user="user!" :handle-click-logout="handleClickLogout" />
+  <main class="section">
+    <div class="container">
+      <div class="tile is-ancestor">
+        <PostList
+          :posts="postList"
+          :chosenPost="chosenPost"
+          :change-chosen-post="changeChosenPost"
+          :close-current-post="closeCurrentPost"
+          :open-create-new-post="openCreateNewPost"
+          :sidebarStatus="sidebarStatus"
+        />
+        <Sidebar
+          :user="user!"
+          :submit-post-change="submitPostChange"
+          :editPost="editPost"
+          :close-post-editing="closePostEditing"
+          :chosen-post="chosenPost"
+          :comments-loading="commentsLoading"
+          :post-comments="postComments"
+          :delete-chosen-post="deleteChosenPost"
+          :sidebar-status="sidebarStatus"
+          :is-comments-editing="isCommentsEditing"
+          :changeCommentAddingState="changeCommentAddingState"
+          :add-new-comment="addNewComment"
+          :deletePostComment="deletePostComment"
```

Comment:
To samo co z robiłes w App.vue mógłbys zrobic tez tutaj. 
Np taka funkcja jak "add-new-comment" nie musi byc przekazywana przez propsa z komponentu PostsPage bo tylko i wyłacznie twój komponent Sidebar tego uzywa. 
Z poziomu PostPage nie dodajesz komentarzy robi to dopiero Sidebar.

Dlatego dla wiekszej czytelnosci kodu lepiej by było umiescic te funkcje tylko i wyłacznie w Sidebar bo PostsPage po prostu to nie obchodzi.

Tak na prawde do Sidebar jedyne co mogłbys przekazywac to np ID postu i wtedy na tej podstawie dodawac mu komentarze edytowac czy usuwac